Description: Permit custom paths to third-party code and submodules
 Do not redefine the third_party_loc and the submodules_loc variables if they
 are set with command line options.
Author: Nicholas Guriev <guriev-ns@ya.ru>
Last-Modified: Thu, 20 Feb 2020 18:46:19 +0300

--- a/cmake/variables.cmake
+++ b/cmake/variables.cmake
@@ -81,5 +81,9 @@ if (NOT APPLE OR build_osx)
 else()
     get_filename_component(libs_loc "../Libraries/macos" REALPATH)
 endif()
-get_filename_component(third_party_loc "Telegram/ThirdParty" REALPATH)
-get_filename_component(submodules_loc "Telegram" REALPATH)
+if (NOT DEFINED third_party_loc)
+    get_filename_component(third_party_loc "Telegram/ThirdParty" REALPATH)
+endif()
+if (NOT DEFINED submodules_loc)
+    get_filename_component(submodules_loc "Telegram" REALPATH)
+endif()
